### PR TITLE
feat: Support for iOS gameloop tests

### LINKED ISSUE #1341: IOS GCloud, Flank command parity
This is a perfectly valid gcloud command:
```
gcloud alpha firebase test ios run 
--type game-loop --app build/ipa/flankgameloop.ipa --scenario-numbers 1 
--device model=iphone11,version=13.6
```

This is not:
```
gcloud:
  app: test_projects/ios/FlankGameLoopExample/ipa/FlankGameLoopExample.ipa
  type: game-loop
  scenario-numbers:
      - 1
  device:
      - model: iphone11
        version: 13.6
        orientation: portrait
        locale: en_US
```

Flank expects a xctestrunfile & test, as it fails the above command with:
`Both of following options must be specified [test, xctestrun-file].`

is this necessary?

some changes would be needed to get it to parity.

### PR DESCRIPTION
**Pull request by adamfilipow92**
Fixes #1341 

1. Run iOS gameloop test 

```
gcloud:
  app: test_projects/ios/FlankGameLoopExample/ipa/FlankGameLoopExample.ipa
  type: game-loop
  scenario-numbers:
      - 1
  device:
      - model: iphone11
        version: 13.6
        orientation: portrait
        locale: en_US
```

2. Test should pass

## Checklist

- [X] Unit tested
- [X] Integration tests updated

### CODE REVIEW

**Comment by adamfilipow92** [comment]
@flank-it

**Comment by adamfilipow92** [comment]
@flank-it

File: test_runner/src/main/kotlin/ftl/args/ValidateIosArgs.kt [hunk 1]
```diff
@@ -89,7 +98,7 @@ private fun IosArgs.validType() {
         throw FlankConfigurationError("Type should be one of ${validIosTypes.joinToString(",")}")
 }
 
-private fun IosArgs.assertXcTestRunData() {
+private fun IosArgs.assertXcTestRunData() = takeIf { isXcTest }?.let {
```
Comment (piotradamczyk5) [inline test_runner/src/main/kotlin/ftl/args/ValidateIosArgs.kt:101]: how about
```kotlin
private fun IosArgs.assertXcTestRunData() =
    takeIf { isXcTest && !disableSharding && testTargets.isNotEmpty() }
        ?.let {
            val filteredMethods = xcTestRunData.filterMethods()

            if (filteredMethods.isEmpty()) throw FlankGeneralError(
                "Empty shards. Cannot match any method to $testTargets"
            )

            if (filteredMethods.size < testTargets.size) {
                val regexList = testTargets.mapToRegex()

                val notMatched = testTargets.filter {
                    filteredMethods.all { method ->
                        regexList.any { regex ->
                            regex.matches(method)
                        }
                    }
                }

                logLn("WARNING: cannot match test_targets: $notMatched")
            }
        }

private fun XcTestRunData.filterMethods() = shardTargets.values
    .flatten()
    .flatMap { it.values }
    .flatten()
```

**Comment by piotradamczyk5** [comment]
Test succeed also on local run >> https://console.firebase.google.com/project/flank-open-source/testlab/histories/bh.9f1d4a7d3ed12f86/matrices/4893467371196578559/executions/bs.3ff65d160e8ad018

**Comment by Sloox** [comment]
@flank-it

**Comment by Sloox** [comment]
@flank-it

File: test_runner/src/main/kotlin/ftl/args/ValidateIosArgs.kt [hunk 2]
```diff
@@ -89,30 +99,31 @@ private fun IosArgs.validType() {
...
+                logLn("WARNING: cannot match test_targets: $notMatched")
+            }
         }
-    }
-}
+
+private fun XcTestRunData.filterMethods() = shardTargets.values
+    .flatten()
+    .flatMap { it.values }
+    .flatten()
```
Comment (axelzuziak-gogo) [inline test_runner/src/main/kotlin/ftl/args/ValidateIosArgs.kt:129]: Flatten everything!

File: test_runner/src/main/kotlin/ftl/gc/GcIosTestMatrix.kt [hunk 3]
```diff
@@ -68,10 +46,10 @@ object GcIosTestMatrix {
             .setDisableVideoRecording(!args.recordVideo)
             .setTestTimeout("${testTimeoutSeconds}s")
             .setIosTestSetup(iOSTestSetup)
-            .setIosXcTest(iOSXCTest)
+            .setupIosTest(iosTestContext)
```
Comment (axelzuziak-gogo) [inline test_runner/src/main/kotlin/ftl/gc/GcIosTestMatrix.kt:49]: Nice logic extraction here 👍

**Review by axelzuziak-gogo — APPROVED** [review]

**Review by piotradamczyk5 — APPROVED** [review]

**Pull request merged**